modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManagerTest.java

```diff
@@ -0,0 +1,1547 @@
+
+    private MeResourceManager meResourceManager;
+    private UserManager userManager;
+
+    @BeforeMethod
+    public void setUp() {
+
+        meResourceManager = new MeResourceManager();
+        userManager = mock(UserManager.class);
+    }
+
+    @AfterMethod
+    public void tearDown() {
+
+    }
+
+    private SCIMResponse getEncodeSCIMExceptionObject(AbstractCharonException exception) {
+
+        JSONEncoder encoder = new JSONEncoder();
+        Map<String, String> responseHeaders = new HashMap<>();
+        responseHeaders.put(SCIMConstants.CONTENT_TYPE_HEADER, SCIMConstants.APPLICATION_JSON);
+        return new SCIMResponse(exception.getStatus(), encoder.encodeSCIMException(exception), responseHeaders);
+    }
+
+    private User getNewUser() throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+
+        return decoder.decodeResource(NEW_USER_SCIM_OBJECT_STRING, schema, new User());
+    }
+
+    @DataProvider(name = "dataForGetSuccess")
+    public Object[][] dataToGetSuccess() throws CharonException, InternalErrorException, BadRequestException {
+
+        User user = getNewUser();
+        String id = user.getId();
+        String name = user.getUserName();
+
+        return new Object[][]{
+                {id, name, null, null, ResponseCodeConstants.CODE_OK, user},
+                {id, name, null, "emails", ResponseCodeConstants.CODE_OK, user},
+                {id, name, "username,meta", null, ResponseCodeConstants.CODE_OK, user},
+                {id, name, "username", "emails", ResponseCodeConstants.CODE_OK, user}
+        };
+    }
+
+    @Test(dataProvider = "dataForGetSuccess")
+    public void testGetSuccess(String id, String name, String attributes,
+                               String excludeAttributes, int expectedScimResponseStatus, Object objectUser)
+            throws BadRequestException, NotFoundException, CharonException {
+
+        User user = (User) objectUser;
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema) CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(ENDPOINT_ME);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(userManager.getMe(name, requiredAttributes)).thenReturn(user);
+
+        SCIMResponse outputScimResponse = meResourceManager.get(name, userManager, attributes, excludeAttributes);
```

Comment:
```suggestion
        SCIMResponse scimResponse = meResourceManager.get(name, userManager, attributes, excludeAttributes);
```

Reply:
Fixed in commit a5699e694bc31369fb4a347d4a0a12177e73775d